From b97b3a2f35f8cf0a9eae975ece510c2bb27ceef9 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Fri, 21 May 2021 15:24:13 +0800
Subject: [PATCH] patch_agent: do not do the packages_iter if pkggrp is None

Add the handling of packages_iter to the else block to avoid:
AttributeError: 'NoneType' object has no attribute 'packages_iter'

Upstream-Status: Inappropriate [poky-stx specific]

Rebased for stx 5.0

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 cgcs-patch/cgcs-patch/cgcs_patch/patch_agent.py | 45 +++++++++++++------------
 1 file changed, 23 insertions(+), 22 deletions(-)

diff --git a/cgcs-patch/cgcs-patch/cgcs_patch/patch_agent.py b/cgcs-patch/cgcs-patch/cgcs_patch/patch_agent.py
index e895e00..8de3c0a 100644
--- a/cgcs-patch/cgcs-patch/cgcs_patch/patch_agent.py
+++ b/cgcs-patch/cgcs-patch/cgcs_patch/patch_agent.py
@@ -536,29 +536,30 @@ class PatchAgent(PatchService):
                 break
 
         if pkggrp is None:
-            LOG.error("Could not find software group: %s", grp_id)
-
-        for pkg in pkggrp.packages_iter():
-            try:
-                res = pkgs_installed.filter(name=pkg.name)
-                if len(res) == 0:
-                    found_pkg = avail.filter(name=pkg.name)
-                    self.missing_pkgs_dnf.append(found_pkg[0])
-                    self.missing_pkgs.append(found_pkg[0].name)
+            LOG.warning("Could not find software group: %s", grp_id)
+        else:
+            for pkg in pkggrp.packages_iter():
+                try:
+                    res = pkgs_installed.filter(name=pkg.name)
+                    if len(res) == 0:
+                        found_pkg = avail.filter(name=pkg.name)
+                        self.missing_pkgs_dnf.append(found_pkg[0])
+                        self.missing_pkgs.append(found_pkg[0].name)
+                        self.changes = True
+                except dnf.exceptions.PackageNotFoundError:
+                    self.missing_pkgs_dnf.append(pkg)
+                    self.missing_pkgs.append(pkg.name)
                     self.changes = True
-            except dnf.exceptions.PackageNotFoundError:
-                self.missing_pkgs_dnf.append(pkg)
-                self.missing_pkgs.append(pkg.name)
-                self.changes = True
-
-        self.installed = self.pkgobjs_to_list(self.installed_dnf)
-        self.to_install = self.pkgobjs_to_list(self.to_install_dnf + self.to_downgrade_dnf)
-
-        LOG.info("Patch state query returns %s", self.changes)
-        LOG.info("Installed: %s", self.installed)
-        LOG.info("To install: %s", self.to_install)
-        LOG.info("To remove: %s", self.to_remove)
-        LOG.info("Missing: %s", self.missing_pkgs)
+
+            self.installed = self.pkgobjs_to_list(self.installed_dnf)
+            self.to_install = self.pkgobjs_to_list(self.to_install_dnf + self.to_downgrade_dnf)
+
+            LOG.info("Patch state query returns %s", self.changes)
+            LOG.info("Installed: %s", self.installed)
+            LOG.info("To install: %s", self.to_install)
+            LOG.info("To remove: %s", self.to_remove)
+            LOG.info("Missing: %s", self.missing_pkgs)
+
         if len(self.duplicated_pkgs) > 0:
             LOG.info("Duplicated: %s", self.duplicated_pkgs)
 
-- 
2.7.4

